pkg/services/secrets/kvstore/plugin.go

```diff
@@ -150,13 +175,15 @@ func (kv *secretsKVStorePlugin) GetAll(ctx context.Context) ([]Item, error) {
 	return parseItems(res.Items), err
 }
 
-func (kv *secretsKVStorePlugin) Fallback() SecretsKVStore {
-	return kv.fallback
+func (kv *SecretsKVStorePlugin) Fallback() SecretsKVStore {
+	return kv.fallbackStore
 }
 
-func (kv *secretsKVStorePlugin) SetFallback(store SecretsKVStore) error {
-	kv.fallback = store
-	return nil
+func (kv *SecretsKVStorePlugin) WithFallbackEnabled(fn func() error) error {
+	kv.fallbackEnabled = true
```

Comment:
maybe use a mutex here to avoid multiple threads changing the fallbackEnabled val?